bin/sirius-script-procedures-machine-shutdown.py

```diff
@@ -0,0 +1,1121 @@
+        if not eghv.wait_voltage(0, timeout=1):
+            self.log('ERR:Timed out waiting for egun high voltage.')
+            return False
+
+        if not self.continue_execution():
+            return False
+
+        self.log('...done. Disabling EGun High Voltage Enable State...')
+        if not eghv.cmd_turn_off(timeout=5):
+            self.log('ERR:Could not disable egun high voltage.')
+            return False
+
+        self.log('...done.')
+        return True
+
+    def s14_start_counter(self):
+        """Check whether the countdown to tunnel access has started."""
+        self.log('Step 14: Checking tunnel access countdown...')
+
+        msg = (
+            'Check whether the countdown to tunnel access has started.')
+        input(msg)
+        # dev = self._devrefs['asppsctrl']
+        # if dev.remaining_time_for_tunnel_access() == 360:
+        #     return False
+        return True
+
+    def s15_disable_ps_triggers(self):
+        """Disable PS triggers."""
+        self.log('Step 15: Disabling PS triggers...')
+        return self._disable_ps_triggers('PS')
+
+    def s16_run_ps_turn_off_procedure(self):
+        """Do turn off PS procedure."""
+        self.log('Step 16: Executing Turn Off PS...')
+        return self._exec_ps_turnoff('PS')
+
+    def s17_disable_pu_triggers(self):
+        """Disable PU triggers."""
+        self.log('Step 17: Disabling PU triggers...')
+        return self._disable_ps_triggers('PU')
+
+    def s18_run_pu_turn_off_procedure(self):
+        """Do turn off PU procedure."""
+        self.log('Step 18: Executing Turn Off PU...')
+        return self._exec_ps_turnoff('PU')
+
+    def execute_procedure(self):
+        """Execute machine shutdown procedure."""
+        steps = (
+            self.s01_close_gamma_shutter,
+            self.s02_macshift_update,
+            self.s03_injmode_update,
+            self.s04_injcontrol_disable,
+            self.s05_ids_parking,
+            self.s06_sofb_fofb_turnoff,
+            self.s07_bbb_turnoff,
+            self.s08_sirf_turnoff,
+            self.s09_borf_turnoff,
+            self.s10_modulators_turnoff,
+            self.s11_adjust_egunbias,
+            self.s12_adjust_egunfilament,
+            self.s13_disable_egun_highvoltage,
+            self.s14_start_counter,
+            self.s15_disable_ps_triggers,
+            self.s16_run_ps_turn_off_procedure,
+            self.s17_disable_pu_triggers,
+            self.s18_run_pu_turn_off_procedure,
+        )
+        for i, step in enumerate(steps):
+            if not step():
+                self.log(f'ERR:Could not complete, failed in step {i:02}.')
+                return False
+            if not self.continue_execution():
+                self.log('ERR:Could not complete, abort received.')
+                return False
+        self.log('Final Step: Wait for tunnel access to be released.')
+        return True
+
+    def _create_devices(self):
```

Comment:
I think we could make use of the the new `Device` interface and only connect with the PVs that will be used here, what do you think? I think it will be worth the effort, at least for the large devices.

Reply:
Agree!! I will do it as soon as possible!

Reply:
done!